### PR TITLE
efi: Add workaround to make it possible to use shim 15.7

### PR DESCRIPTION
**Pull request by chrisccoulson**
This is a temporary solution which should make it possible to work with shim 15.7. It corrects a typo in one of the host SbatLevel payloads, and also adds a workaround for a bug introduced into shim 15.7. It borrows a little bit of code from a PR for master (shimVersion).

### CODE REVIEW

**Comment by chrisccoulson** [comment]
Note that this makes all of the tests fail because the test binaries don't have the right section in them.

**Review by mvo5 — APPROVED** [review]
Thanks, this looks reasonable to me

File: efi/secureboot_policy.go [hunk 1]
```diff
@@ -892,6 +969,14 @@ func (g *secureBootPolicyGen) processShimExecutableLaunch(branches []*secureBoot
 		flags |= shimVariableAuthorityEventsMatchSpec
 	}
 
+	version, err := shim.version()
+	if err != nil {
+		return xerrors.Errorf("cannot determine shim version: %w", err)
+	}
+	if version.Compare(shimVersion{15, 7}) >= 0 {
```
Comment (pedronis) [inline efi/secureboot_policy.go:976]: as discussed, a gadget with a fixed shim will need to assume a snapd that stops doing this or produces branches both for the fixed shim and the buggy shim

**Review by pedronis — APPROVED** [review]
lgtm

**Comment by mvo5** [comment]
This was tested now on real devices and seems to work fine so I will merge it.

**Pull request merged**